index.ts

```diff
@@ -1,35 +1,60 @@
 export class OatyArray {
-  private _original: object[]
-  private _keys: string[]
-  private _transposed: object = {}
+  private _keys: string[] | undefined
+  private _original: object[] = []
+  private _transposed: { [key: string]: any } = {}
+
+  constructor(data?: object[], options?: { keys?: string[] }) {
+    if (data !== undefined) { this.push(...data) }
+    if (options !== undefined) {
+      if (options.keys !== undefined) { this._keys = options.keys }
+    }
+  }
+
+  get keys(): string[] | undefined {
+    return this._keys
+  }
 
   get length(): number {
     return this._original.length
   }
 
-  constructor(input: object[], keys: string[]) {
-    this._original = input.map((x) => Object.assign({}, x))
-    this._keys = keys
+  get original(): object[] {
+    return this._original
+  }
 
-    this._transposed = transpose(this._keys, this._original)
+  get transposed(): object[] {
+    return this._original
   }
 
-  public get(keyName: string, keyValue: string): object[] {
+  public get(keyName: string, keyValue: string): object[] | undefined {

```

Comment:
this whole function can just be

```ts
return this._transposed[keyName] ? this._transposed[keyName][keyValue] : undefined
```

Reply:
Was more in preparation for the future when we'll add different returns (undefined, [], throw) for when an object isn't found.

Reply:
"Premature optimization is the root of all evil" - Donald Knuth

Reply:
I agree but it'll be the next thing I do after everything on this review as it's something I need

Reply:
Outdated